src/libecalc/presentation/yaml/mappers/consumer_function_mapper.py

```diff
@@ -1141,6 +1080,73 @@ def _map_compressor(
             else None
         )
 
+        # Determine model type and create appropriately - key integration point
+        model_ref = model.energy_function
+        yaml_model = self.__references.get_compressor_model(model_ref)
```

Comment:
```suggestion
        yaml_model = self.__references.get_compressor_model(model.energy_function)
```

directly? not used other places? avoid unneccessary difficulty in "naming it"?

Reply:
fixed